x-pack/plugins/ingest_manager/public/plugin.ts

```diff
@@ -78,18 +75,19 @@ export class IngestManagerPlugin
   }
 
   public async start(core: CoreStart): Promise<IngestManagerStart> {
-    try {
-      const permissionsResponse = await core.http.get(appRoutesService.getCheckPermissionsPath());
-      if (permissionsResponse.success) {
-        const { isInitialized: success } = await core.http.post(setupRouteService.getSetupPath());
-        return { success, registerDatasource };
-      } else {
-        throw new Error(permissionsResponse.error);
-      }
-    } catch (error) {
+    const permissionsResponse = await core.http.get<CheckPermissionsResponse>(
+      appRoutesService.getCheckPermissionsPath()
+    );
+    if (permissionsResponse.success) {
+      const successPromise = core.http
```

Comment:
@rudolf Since this no longer `await`s, would it be ok to move it to `setup`? Any reason from a Platform POV not t start the async/http work as early as possible if it doesn't block?